### PR TITLE
Add sidebar to documentation

### PR DESCRIPTION
**Pull request by driehle**

### CODE REVIEW

File: docs/en/sidebar.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.. toctree::
+    :depth: 3
+
+    Introduction <index>
```
Comment (greg0ire) [inline docs/en/sidebar.rst]: What does this `<index>` syntax do? I've looked at other `sidebar.rst` files across repositories, and I don't see it.

Comment (SenseException) [reply docs/en/sidebar.rst]: I'm checking if this is working with the current rst-parser

Comment (driehle) [reply docs/en/sidebar.rst]: It should name the section different in the TOC. See "Entries" in https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html. Though I don't know if the Doctrine parser understands it. Thanks for checking this, Claudio!

Comment (driehle) [reply docs/en/sidebar.rst]: https://github.com/doctrine/DoctrineMongoODMModule/pull/237 shows that this doesn't work unfortunately, the item simply disapears from the website, see https://www.doctrine-project.org/projects/doctrine-mongo-odm-module/en/3.0/index.html. I updated this PR accordingly. We'll simply name the first heading in `index.rst` then "Introduction". This is also done in other Doctrine projects.

Comment (SenseException) [reply docs/en/sidebar.rst]: Sadly it doesn't work.

**Review by SenseException — APPROVED** [review]

**Pull request merged**